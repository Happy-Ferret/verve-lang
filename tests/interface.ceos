interface show_<t> {
  virtual show(t) -> string
}

implementation show_<string> {
  fn show(s) {
    "SHOW STRING"
  }
}

implementation show_<int> {
  fn show(n) {
    "SHOW INT"
  }
}

print(show(""))
print(show(0))
