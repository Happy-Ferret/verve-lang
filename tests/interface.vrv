interface show_<t> {
  virtual show(t) -> string

  fn foo(s: t) -> string {
    "FOO DEFAULT"
  }
}

implementation show_<string> {
  fn show(s) {
    "SHOW STRING"
  }
}

implementation show_<int> {
  fn show(n) {
    "SHOW INT"
  }

  fn foo(s) {
    "FOO INT"
  }
}

print(show(""))
print(show(0))

print(foo(""))
print(foo(0))
