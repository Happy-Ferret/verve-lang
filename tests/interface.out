SHOW STRING
SHOW INT
FOO DEFAULT
FOO INT
